### ISSUE TITLE
Upgrade to algolia/algoliasearch-client-php v3

### ISSUE THREAD

**Issue by earthwytch** [wilr/silverstripe-algolia#54]
Is it possible to upgrade algolia/algoliasearch-client-php to version 3? 

I'm getting errors since I upgraded to PHP8.1 similar to this:
```
Return type of Algolia\AlgoliaSearch\Response\AbstractResponse::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in vendor/algolia/algoliasearch-client-php/src/Response/AbstractResponse.php on line 34
```
Version 3 of the algolia search client seems to fix this by adding the `#[\ReturnTypeWillChange]` directive in /src/Response/AbstractResponse.php. Not sure what this involves for you or the flow-on effects but thought I'd ask 🙂

**Comment by mikey-harveycameron** [comment]
yeah this has killed my site.

**Comment by wilr** [comment]
Fixed in 1.4.12
